examples/chat/client/tasks.nim

```diff
@@ -255,34 +295,36 @@ proc startWakuChat2*(username: string) {.task(kind=no_rts, stoppable=false).} =
 
   nick = username
 
-  when PayloadV1:
-    # generate `symKey` here instead of `statusContext` to decrease startup
-    # time of `TaskRunner` instance and therefore time to first paint of TUI
-    if not symKeyGenerated:
-      symKey = generateSymKey(contentTopic)
-      symKeyGenerated = true
-
-  let (extIp, extTcpPort, extUdpPort) = setupNat(conf.nat, clientId,
-    Port(uint16(conf.tcpPort) + conf.portsShift),
-    Port(uint16(conf.udpPort) + conf.portsShift))
-
-  var nodekey: waku_chat2.crypto.PrivateKey
-
-  if $conf.nodekey == "":
-    nodekey = waku_chat2.crypto.PrivateKey.random(Secp256k1,
-      waku_chat2.keys.newRng()[]).tryGet()
-  else:
-    nodekey = waku_chat2.crypto.PrivateKey(scheme: Secp256k1,
-      skkey: SkPrivateKey.init(
-        waku_chat2.utils.fromHex($conf.nodekey)).tryGet())
+  # invoke `setupNat` here instead of `statusContext` to decrease startup time
```

Comment:
The `startWakuChat2` proc is massive, and we should definitely figure out a way to split this up.

Reply:
Yes, it should be broken up *and* this logic should really be driven by nim-status as opposed to being a sibling of nim-status logic inside `examples/chat/client/tasks.nim`.